lib/internal/mime.js

```diff
@@ -16,62 +16,140 @@ const {
+    result += charCode >= ASCII_LOWER_LOWERBOUND &&
+      charCode <= ASCII_LOWER_UPPERBOUND ?
       StringPrototypeToLowerCase(char) :
       char;
   }
   return result;
 }
 
-const SOLIDUS = '/';
-const SEMICOLON = ';';
+// https://fetch.spec.whatwg.org/#collect-an-http-quoted-string
+function collectHTTPQuotedString(str, marker) {
+  const limit = str.length;
+  let index = marker.position;
+  // #2 Let value be the empty string.
+  let value = '';
+
+  // #4 Advance position by 1.
+  index++;
+
+  // #5 While true
+  while (true) {
+    let nextDelimiter = StringPrototypeIndexOf(str, BACKSLASH, index);
+
+    nextDelimiter = nextDelimiter !== -1 ? nextDelimiter : StringPrototypeIndexOf(str, QUOTE, index);
+
+    // #5-1 Append the result of collecting a sequence of code points
+    // that are not U+0022 (") or U+005C (\) from input, given position, to value.
+    const collectedPoints = nextDelimiter === -1 ?
+      StringPrototypeSlice(str, index) : StringPrototypeSlice(str, index, nextDelimiter);
+    index += collectedPoints.length;
+    value += collectedPoints;
+
+    // #5-2 If position is past the end of input, then break.
+    if (index > limit) break;
+
+    // #5-3 Let quoteOrBackslash be the code point at position within input.
+    const quoteOrBackslash = str[index];
+    const quoteOrBackslashAsCode = quoteOrBackslash == null ?
+      0 :
+      StringPrototypeCharCodeAt(quoteOrBackslash, 0);
+
+    // #5-4 Advance position by 1.
+    index++;
+
+    // #5-5 If quoteOrBackslash is U+005C (\), then:
+    if (quoteOrBackslashAsCode === BACKSLASH_CHARCODE) {
+      // #5-5-1 If position is past the end of input, then append U+005C (\) to value and break.
+      if (index >= limit) {
+        value += BACKSLASH;
+        break;
+      }
+
+      // #5-5-2 Append the code point at position within input to value.
+      value += str[index];
+
+      // #5-5-3 Advance position by 1.
+      index++;
+    } else if (quoteOrBackslashAsCode === QUOTE_CHARCODE) {
+      // #5-6 Otherwise
+      // #5-6-1 Assert: quoteOrBackslash is U+0022 (").
+      // #5-6-2 Break
+      break;
+    }
+  }
+
+  marker.position = index;
+
+  // #6 If the extract-value flag is set, then return value
+  return value;
+}
+
 function parseTypeAndSubtype(str) {
-  // Skip only HTTP whitespace from start
-  let position = SafeStringPrototypeSearch(str, END_BEGINNING_WHITESPACE);
-  // read until '/'
-  const typeEnd = StringPrototypeIndexOf(str, SOLIDUS, position);
-  const trimmedType = typeEnd === -1 ?
-    StringPrototypeSlice(str, position) :
+  const strLength = str.length;
+  // #2 Let position be a position variable for input, initially pointg at
+  // the start of the input
+  let position = 0;
+
+  // #3 Let type be the result of collecting a sequence of code points that are
+  // not U+002F (/ [SOLIDUS]) from input, given position.
+  const typeEnd = StringPrototypeIndexOf(str, SOLIDUS);
+  const preType = typeEnd === -1 ?
```

Comment:
is branchless maybe faster?

Reply:
Better for speculation, sure. What do you have in mind?